src/Orleans.Core/Streams/Internal/StreamImpl.cs

```diff
@@ -113,6 +118,14 @@ public Task OnErrorAsync(Exception ex)
             return GetConsumerInterface().ResumeAsync(handle, observer, token);
         }
 
+        internal Task<StreamSubscriptionHandle<T>> ResumeAsync(
```

Comment:
what would happen if a user used both ResumeAsync method.

Reply:
Last resume call should win.  This is because it creates a new handle and the new handle will only have the most recent observable.  If both are called on the same handle one should fail because the handle should be invalidated after new handle is created.